scio-google-cloud-platform/src/it/scala/com/spotify/scio/bigquery/BigQueryIT.scala

```diff
@@ -37,14 +38,24 @@ object BigQueryIT {
   class WordCount
 }
 
-class BigQueryIT extends AnyFlatSpec with Matchers {
+class BigQueryIT extends AnyFlatSpec with Matchers with BeforeAndAfterEach with BeforeAndAfterAll {
   import BigQueryIT._
 
   // =======================================================================
   // Integration test with mock data
   // =======================================================================
 
-  "MockBigQuery" should "support mock data" in {
+  val StagingDatasetPrefix = "scio_bigquery_staging_custom"
+
+  override def beforeEach() {
+    sys.props.subtractOne(BigQuerySysProps.StagingDatasetPrefix.flag)
+  }
+
+  override def afterAll() {
```

Comment:
Instead of overriding `beforeEach` and `afterAll` to set and unset the property, I think it would be simpler to just add this setting/unsetting logic to the beginning/end of the `it should "support mock data with custom staging dataset"` unit test 👍 If we were testing this property in more than one unit test,`beforeEach` would make sense, but I vote for keeping it simple for just the single unit test :)

otherwise, this looks good!

Reply:
Two tests use it :) I don't think beginning and after each test is a good idea because in case an exception happens during the test, it won't clean after itself. This is more future proof, as it resets the state before each test, regardless of whether they care about the property or not. and one clean up after the entire suite.